### ISSUE TITLE
Fan Information

### ISSUE THREAD

**Issue by felipebraz** [sebhildebrandt/systeminformation#210]
Hi... Unfortunelly for some reason I can´t read my CPU Temperature... But would be awesome if I could at least see my fans (chassis, CPU, GPU...)...

Thanks!!!

**Comment by sebhildebrandt** [comment]
@felipebraz, two questions:

- did you read the **Known Issues** section in the readme?
- what OS are you running?

**Comment by felipebraz** [comment]
Yes I did... This is why I'm not complaining about the temperature issue, it seems my pc is presenting problems to show this info... My OS is Windows 10.

What I really desire, knowing the temperature won't work, is read the fan speed. Is it possible in a future?

Enviado do meu iPhone

Em 17 de mar de 2019, à(s) 17:59, Sebastian Hildebrandt <notifications@github.com<mailto:notifications@github.com>> escreveu:


@felipebraz<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffelipebraz&data=02%7C01%7C%7Ca1f77c93a6c044867b9708d6ab1b69d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636884531553690043&sdata=iScULdRtSQgcqwTkLhE3lENGbRPt8wcYdXL1pg5eelQ%3D&reserved=0>, two questions:

  *   did you read the Known Issues section in the readme?
  *   what OS are you running?

**Comment by sebhildebrandt** [comment]
@felipebraz ... yes this is a nice idea. Will make a research on that.

**Comment by PCGeekBrain** [comment]
@sebhildebrandt If it helps at all I did a bit of research for windows and it appears that there is a Win32 library for getting fan speed.

Here is a sample sciript: http://eddiejackson.net/data/powershell/Get-Fan.ps1 (copied to GitHub: https://gist.github.com/PCGeekBrain/f36eb18b73e02412e3654d3514f463ae) that seems to work just fine on my machine. I don't know if you want to include power-shell scripts in your repo as that would remove it's JS only nature...

Thank you for making this awesome library! Hope to use it in a personal project soon ;-)

**Comment by sebhildebrandt** [comment]
@PCGeekBrain : thank you for pointing me to this script. This will definitely help to implement this. Already having some ideas ;-) I will also implement fan speed for linux ... please give me some time to finish this.

**Comment by PCGeekBrain** [comment]
Of course, no pressure, just here and trying to help.

I dual boot my system at home. I'll try to find a way to get the fan information from Ubuntu as well... Sadly though I do not own a mac...

**Comment by Lonerider2010** [comment]
Hello, just found this issue. Is the fan information implemented now? I am interested in it for Linux arm64.
Thank you.

**Comment by sebhildebrandt** [comment]
@Lonerider2010 @felipebraz Please have a look at the new version 6 (currently in beta). This issue should be solved in the new version. I would be happy if you can test it on your side. Please refer also to the documentation https://systeminformation.io/v6/ and https://systeminformation.io/v6/changes.html  

If you have time to run the full tests and/or single test on your machines, that would be awesome. Here you can find some instructions: [https://systeminformation.io/v6/tests.html](https://systeminformation.io/v6/tests.html)
